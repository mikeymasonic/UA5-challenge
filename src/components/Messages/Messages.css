@font-face {
  font-family: 'Times New Roman PS Std';
  src: url('//db.onlinewebfonts.com/t/d1cd521cd5ec28ada84c431eadc8fe0c.eot');
  src: url('//db.onlinewebfonts.com/t/d1cd521cd5ec28ada84c431eadc8fe0c.eot?#iefix')
      format('embedded-opentype'),
    url('//db.onlinewebfonts.com/t/d1cd521cd5ec28ada84c431eadc8fe0c.woff2')
      format('woff2'),
    url('//db.onlinewebfonts.com/t/d1cd521cd5ec28ada84c431eadc8fe0c.woff')
      format('woff'),
    url('//db.onlinewebfonts.com/t/d1cd521cd5ec28ada84c431eadc8fe0c.ttf')
      format('truetype'),
    url('//db.onlinewebfonts.com/t/d1cd521cd5ec28ada84c431eadc8fe0c.svg#Times New Roman PS Std')
      format('svg');
}

* {
  font-family: 'Times New Roman PS Std';
}

.Slider {
  --slider-height-percentage: 50%;
  --slider-transition-duration: 700ms;
  --organic-arrow-thickness: 4px;
  --organic-arrow-border-radius: 0px;
  --organic-arrow-height: 40px;
  --organic-arrow-color: #26456f;
  --control-button-width: 10%;
  --control-button-height: 25%;
  --control-button-background: transparent;
  --control-bullet-color: #999999;
  --control-bullet-active-color: black;
  --loader-bar-color: #f1d34a;
  --loader-bar-height: 6px;
  --content-background-color: white;
}

.TitleTop {
  font-size: 1.25rem;
  font-weight: bold;
  position: absolute;
  z-index: 100;
  display: flex;
  justify-content: center;
  margin: 1rem auto 0;
  text-transform: uppercase;
  left: 50%;
  transform: translateX(-50%);
}

.TitleBottom {
  font-size: 1.25rem;
  font-weight: bold;
  position: absolute;
  z-index: 100;
  display: flex;
  justify-content: center;
  margin: 1rem auto 0;
  text-transform: uppercase;
  left: 50%;
  transform: translateX(-50%);
}

.Button {
  z-index: 3;
  color: white;
  background-color: black;
  padding: 0.5rem;
  position: relative;
  display: block;
  justify-content: center;
  margin: 0 auto 2rem;
  font-size: 1rem;
}

@media screen and (max-width: 800px) {
  .TitleTop {
    font-size: 0.5rem;
  }

  .TitleBottom {
    font-size: 0.5rem;
  }

  .Button {
    padding: 0.5rem;
    font-size: 0.5rem;
  }
}
