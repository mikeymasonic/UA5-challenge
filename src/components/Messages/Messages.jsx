import React from 'react';
import { messages, moreMessages } from '../../data/data';
import MessagesList from '../MessagesList/MessagesList';
import AwesomeSlider from 'react-awesome-slider';
import AwesomeSliderStyles from '../../styles/styles.scss';
import styles from './Messages.css';

const Messages = () => {
  const titleTop = messages[0].section_title;
  let titleBottom = moreMessages[0].section_title;
  titleBottom = titleBottom.replace('!', '');

  const sectionTitleTop = messages[0].section_title;
  const sectionTitleBottom = moreMessages[0].section_title;

  const messageNodes = messages[0].quotes.map((data) => {
    return (
      <div
        key={data.attribution}
        data-src="https://pbs.twimg.com/profile_images/524717947931996160/g0kMdxBX.png"
      >
        <MessagesList
          key={data.attribution}
          {...data}
          sectionTitle={sectionTitleTop}
        />
      </div>
    );
  });

  const moreMessageNodes = moreMessages[0].quotes.map((data) => {
    return (
      <div
        key={data.attribution}
        data-src="https://pbs.twimg.com/profile_images/524717947931996160/g0kMdxBX.png"
      >
        <MessagesList
          key={data.attribution}
          {...data}
          sectionTitle={sectionTitleBottom}
        />
      </div>
    );
  });
  return (
    <>
      <section className={styles.Messages}>
        <section className={styles.Top}>
          <h1 className={styles.Title}>{titleTop}</h1>
          <AwesomeSlider
            className={styles.Slider}
            scssModule={AwesomeSliderStyles}
          >
            {messageNodes}
          </AwesomeSlider>
        </section>
        <section className={styles.Bottom}>
          <h1 className={styles.Title}>{titleBottom}</h1>
          <AwesomeSlider
            className={styles.Slider}
            scssModule={AwesomeSliderStyles}
          >
            {moreMessageNodes}
          </AwesomeSlider>
        </section>
      </section>
    </>
  );
};

export default Messages;
