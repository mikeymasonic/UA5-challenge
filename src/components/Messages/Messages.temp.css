@font-face {
  font-family: 'Times New Roman PS Std';
  src: url('//db.onlinewebfonts.com/t/d1cd521cd5ec28ada84c431eadc8fe0c.eot');
  src: url('//db.onlinewebfonts.com/t/d1cd521cd5ec28ada84c431eadc8fe0c.eot?#iefix')
      format('embedded-opentype'),
    url('//db.onlinewebfonts.com/t/d1cd521cd5ec28ada84c431eadc8fe0c.woff2')
      format('woff2'),
    url('//db.onlinewebfonts.com/t/d1cd521cd5ec28ada84c431eadc8fe0c.woff')
      format('woff'),
    url('//db.onlinewebfonts.com/t/d1cd521cd5ec28ada84c431eadc8fe0c.ttf')
      format('truetype'),
    url('//db.onlinewebfonts.com/t/d1cd521cd5ec28ada84c431eadc8fe0c.svg#Times New Roman PS Std')
      format('svg');
}

* {
  font-family: 'Times New Roman PS Std';
}

.Slider {
  --slider-height-percentage: 60%;
  --slider-transition-duration: 700ms;
  --organic-arrow-thickness: 4px;
  --organic-arrow-border-radius: 0px;
  --organic-arrow-height: 40px;
  --organic-arrow-color: #26456f;
  --control-button-width: 10%;
  --control-button-height: 25%;
  --control-button-background: transparent;
  --control-bullet-color: #999999;
  --control-bullet-active-color: black;
  --loader-bar-color: #f1d34a;
  --loader-bar-height: 6px;
  --content-background-color: white;
}

.Title {
  font-size: 1.25rem;
  font-weight: bold;
  position: relative;
  top: 10rem;
  z-index: 100;
  display: flex;
  justify-content: center;
  margin: auto;
  text-transform: uppercase;
}

.Top {
  margin-top: -7rem;
}

.Bottom {
  margin-top: -8rem;
}

.Button {
  z-index: 3;
  color: white;
  background-color: black;
  height: 3rem;
  width: 8rem;
  position: relative;
  display: block;
  justify-content: center;
  margin: auto;
  top: -9rem;
  font-size: 1rem;
}
