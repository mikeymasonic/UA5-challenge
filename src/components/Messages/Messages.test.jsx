import React from 'react';
import { shallow } from 'enzyme';
import Messages from './Messages';

describe('Messages component', () => {
  it('renders Messages', () => {
    const wrapper = shallow(<Messages />);
    expect(wrapper).toMatchSnapshot();
  });
});
