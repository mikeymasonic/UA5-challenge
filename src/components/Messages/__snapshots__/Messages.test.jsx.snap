// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages component renders Messages 1`] = `
<Fragment>
  <section
    className="Messages"
  >
    <section
      className="Top"
    >
      <h1
        className="Title"
      >
        Congratulatory Messages
      </h1>
      <AwesomeSlider
        animation={null}
        bullets={true}
        buttonContentLeft={null}
        buttonContentRight={null}
        buttons={true}
        className="Slider"
        controlsReturnDelay={0}
        cssModule={null}
        customContent={null}
        disabled={false}
        fillParent={false}
        infinite={true}
        media={Array []}
        mobileTouch={true}
        name="awesome-slider"
        onFirstMount={null}
        onLoadStart={null}
        onResetSlider={null}
        onStartupRelease={null}
        onTransitionEnd={null}
        onTransitionRequest={null}
        onTransitionStart={null}
        organicArrows={true}
        rootElement="awssld"
        scssModule={null}
        selected={0}
        startup={true}
        startupDelay={0}
        startupScreen={null}
        style={Object {}}
        transitionDelay={0}
      >
        <div
          data-src="https://pbs.twimg.com/profile_images/524717947931996160/g0kMdxBX.png"
          key="Giuseppe Mcknight
Executive and Artistic Director, UCLA’s Center for the Art of Performance"
        >
          <MessagesList
            attribution="Giuseppe Mcknight
Executive and Artistic Director, UCLA’s Center for the Art of Performance"
            key="Giuseppe Mcknight
Executive and Artistic Director, UCLA’s Center for the Art of Performance"
            message="...please live the beautiful life that you have. It will always be enough..."
            sectionTitle="Congratulatory Messages"
          />
        </div>
        <div
          data-src="https://pbs.twimg.com/profile_images/524717947931996160/g0kMdxBX.png"
          key="Saskia Barrett
Shapiro Director, Fowler Museum"
        >
          <MessagesList
            attribution="Saskia Barrett
Shapiro Director, Fowler Museum"
            key="Saskia Barrett
Shapiro Director, Fowler Museum"
            message="I hope that each one of you will harness your unique abilities, the strength of your convictions, the resiliency of youth, and the fellowship of community—the one at UCLA and others—to launch your next chapter with verve."
            sectionTitle="Congratulatory Messages"
          />
        </div>
        <div
          data-src="https://pbs.twimg.com/profile_images/524717947931996160/g0kMdxBX.png"
          key="Dewey Shelton
Director, Hammer Museum "
        >
          <MessagesList
            attribution="Dewey Shelton
Director, Hammer Museum "
            key="Dewey Shelton
Director, Hammer Museum "
            message="Hello Class of 2020...I don’t need to tell you that you’re graduating at an extraordinary time...I cannot wait to see what you all make, and invent, and design, and solve...so for each and every one of you I offer my congratulations..."
            sectionTitle="Congratulatory Messages"
          />
        </div>
        <div
          data-src="https://pbs.twimg.com/profile_images/524717947931996160/g0kMdxBX.png"
          key="Leela Bowler
Professor, Department of World Arts and Cultures / Dance"
        >
          <MessagesList
            attribution="Leela Bowler
Professor, Department of World Arts and Cultures / Dance"
            key="Leela Bowler
Professor, Department of World Arts and Cultures / Dance"
            message="Creativity is not part of the solution, it IS the solution. Reimagine, reinvent, insist on dignity, and save the world.”"
            sectionTitle="Congratulatory Messages"
          />
        </div>
      </AwesomeSlider>
    </section>
    <section
      className="Bottom"
    >
      <h1
        className="Title"
      >
        More Messages
      </h1>
      <AwesomeSlider
        animation={null}
        bullets={true}
        buttonContentLeft={null}
        buttonContentRight={null}
        buttons={true}
        className="Slider"
        controlsReturnDelay={0}
        cssModule={null}
        customContent={null}
        disabled={false}
        fillParent={false}
        infinite={true}
        media={Array []}
        mobileTouch={true}
        name="awesome-slider"
        onFirstMount={null}
        onLoadStart={null}
        onResetSlider={null}
        onStartupRelease={null}
        onTransitionEnd={null}
        onTransitionRequest={null}
        onTransitionStart={null}
        organicArrows={true}
        rootElement="awssld"
        scssModule={null}
        selected={0}
        startup={true}
        startupDelay={0}
        startupScreen={null}
        style={Object {}}
        transitionDelay={0}
      >
        <div
          data-src="https://pbs.twimg.com/profile_images/524717947931996160/g0kMdxBX.png"
          key="Jae Rigby
Professor, Department of World Arts and Cultures / Dance"
        >
          <MessagesList
            attribution="Jae Rigby
Professor, Department of World Arts and Cultures / Dance"
            key="Jae Rigby
Professor, Department of World Arts and Cultures / Dance"
            message="Creativity is not part of the solution, it IS the solution. Reimagine, reinvent, insist on dignity, and save the world."
            sectionTitle="More Messages!"
          />
        </div>
        <div
          data-src="https://pbs.twimg.com/profile_images/524717947931996160/g0kMdxBX.png"
          key="Keanan Neale
M.F.A., Design Media Arts, ’06"
        >
          <MessagesList
            attribution="Keanan Neale
M.F.A., Design Media Arts, ’06"
            key="Keanan Neale
M.F.A., Design Media Arts, ’06"
            message="Class of 2020. Congratulations. The next chapter is in your hands. Stay thoughtful, act passionately, and remain curious."
            sectionTitle="More Messages!"
          />
        </div>
        <div
          data-src="https://pbs.twimg.com/profile_images/524717947931996160/g0kMdxBX.png"
          key="Dewey Shelton
Director, Hammer Museum "
        >
          <MessagesList
            attribution="Dewey Shelton
Director, Hammer Museum "
            key="Dewey Shelton
Director, Hammer Museum "
            message="Hello Class of 2020...I don’t need to tell you that you’re graduating at an extraordinary time...I cannot wait to see what you all make, and invent, and design, and solve...so for each and every one of you I offer my congratulations..."
            sectionTitle="More Messages!"
          />
        </div>
      </AwesomeSlider>
    </section>
  </section>
</Fragment>
`;
